### PR TITLE
ray-operator: disallow pod creation in namespaces outside of RayCluster namespace

### PR DESCRIPTION
**Pull request by andrewsykim**
## Why are these changes needed?

Currently it is possible to specify a namespace in the pod template that is different from the namespace of the RayCluster. I haven't found a use-case where this is needed. This PR disallows namespaces for Pods outside the namespace of the RayCluster.

## Checks

- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by vinayakankugoyal** [comment]
Do any release notes need to be update?

File: ray-operator/controllers/ray/common/pod.go
Comment (vinayakankugoyal) [inline ray-operator/controllers/ray/common/pod.go:97]: Maybe its better to return an error in this case overriding user values seems wrong?

Comment (vinayakankugoyal) [inline ray-operator/controllers/ray/common/pod.go:97]: could you add a comment here explaining why we are overriding the namespace.

**Comment by vinayakankugoyal** [comment]
The testing section of the PR description needs to be updated. How was this tested?

**Comment by andrewsykim** [comment]
> The testing section of the PR description needs to be updated. How was this tested?

I was waiting for feedback from @kevin85421 before writing tests. I've added tests now. Thanks

File: ray-operator/controllers/ray/common/pod.go
Comment (andrewsykim) [reply ray-operator/controllers/ray/common/pod.go:99]: sure, added

**Review by kevin85421 — APPROVED** [review]
LGTM. @vinayakankugoyal Do you have any other comments? I will merge this PR after your approval.

**Review by vinayakankugoyal — APPROVED** [review]

**Pull request merged**
